wireless: fix EAP inconsistencies; add missing method names (bsc#936514, boo#927615)

pwieczorkiewicz opened the pull request.

mtomaschewski commented:
When I see correctly, this pull changes things in an incompatible way. WIRELESS_EAP_MODE defaults to "TTLS PEAP TLS", otherwise it provides the list. WIRELESS_EAP_AUTH is defined
to be used to set phase2="auth=$WIRELESS_EAP_AUTH", unset by default.

pwieczorkiewicz commented:
Yes, the inconsistency came from the ifcfg.template and documentation I found.
Now submitted corrected version based strictly on ifup-wireless code from SLE11.

New things:
1. Fixed default handling of WIRELESS_EAP_MODE (TTLS PEAP TLS sent to wpa_supplicant)
2. Fixed default handling of WIRELESS_EAP_AUTH ('auth=any' sent to wpa_supplicant)
3. Implementation of phase1 options handling (peapver and peaplabel sent to wpa_supplicant)
4. Proper setting of auth-proto (WPA1, WPA2 or both) for EAP
